sdk/logs/src/main/java/io/opentelemetry/sdk/logs/ExtendedSdkLogRecordData.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.logs;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.api.common.Value;
+import io.opentelemetry.api.incubator.common.ExtendedAttributes;
+import io.opentelemetry.api.logs.Severity;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
+import io.opentelemetry.sdk.logs.data.internal.ExtendedLogRecordData;
+import io.opentelemetry.sdk.resources.Resource;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+@AutoValue
+@AutoValue.CopyAnnotations
+@Immutable
+abstract class ExtendedSdkLogRecordData implements ExtendedLogRecordData {
+
+  ExtendedSdkLogRecordData() {}
+
+  static ExtendedSdkLogRecordData create(
+      Resource resource,
+      InstrumentationScopeInfo instrumentationScopeInfo,
+      @Nullable String eventName,
+      long epochNanos,
+      long observedEpochNanos,
+      SpanContext spanContext,
+      Severity severity,
+      @Nullable String severityText,
+      @Nullable Value<?> body,
+      ExtendedAttributes attributes,
+      int totalAttributeCount) {
+    return new AutoValue_ExtendedSdkLogRecordData(
+        resource,
+        instrumentationScopeInfo,
+        epochNanos,
+        observedEpochNanos,
+        spanContext,
+        severity,
+        severityText,
+        totalAttributeCount,
+        eventName,
+        attributes,
+        body);
+  }
+
+  @Override
+  @Nullable
+  public abstract Value<?> getBodyValue();
+
+  @Override
+  @SuppressWarnings("deprecation") // Implementation of deprecated method
+  public io.opentelemetry.sdk.logs.data.Body getBody() {
```

Comment:
Certainly doesn't have to be in this PR, but I feel like this has been deprecated long enough for us to probably remove now eh?

Reply:
Its has to stay around in the parent `LogRecordData` interface until a 2.x.x given our strong backwards compatibility guarantees. 

It shows up here because of oddities with auto value.

Reply:
I don't love it, but I hear ya. :)